client/signup/steps/domains/index.jsx

```diff
@@ -62,11 +61,39 @@ class DomainsStep extends React.Component {
 		store: PropTypes.object,
 	};
 
-	state = { products: productsList.get() };
+	constructor( props ) {
+		super( props );
+
+		this.skipRender = false;
+
+		const domain = get( props, 'queryObject.new', false );
+		if (
+			this.isDomainsFirstFlow() &&
+			domain &&
+			// If someone has a better idea on how to figure if the user landed anew
+			// Because we persist the signupDependencies, but still want the user to be able to go back to search screen
+			props.path.indexOf( '?' ) !== -1
```

Comment:
It's not elegant, but it works and I can't think of anything else off the top of my head.